xaitk_saliency/interfaces/vis_sal_image_classifier_blackbox.py

```diff
@@ -0,0 +1,99 @@
+import abc
+
+import numpy as np
+from smqtk_classifier import ClassifyImage
+from smqtk_core import Plugfigurable
+
+from xaitk_saliency.exceptions import ShapeMismatchError
+
+
+class GenerateImageClassifierBlackboxSaliency (Plugfigurable):
+    """
+    Interface for algorithms that take a reference image and a image classifier
+    blackbox algorithm, and generates a number of visual saliency heatmap
+    matrices, one for each class output by the classifier blackbox.
+
+    An input classifier blackbox needs to be input, which requires some
+    specification in how to operate the blackbox.
+    The `smqtk_classifier.ClassifyImage` abstract interface is utilized for
+    this purpose in order to provide some minimal form that a blackbox
+    classifier must have: be able to classify an image into confidences for
+    some number of class labels.
+
+    Generated visual saliency heat-map for each input class as a
+    float-type `numpy.ndarray` of shape `[nClasses x H x W]`.
+    """
+
+    def generate(
+        self,
+        ref_image: np.ndarray,
+        blackbox: ClassifyImage
+    ) -> np.ndarray:
+        """
+        Generate per-class visual saliency heatmaps for some classifier
+        blackbox over some image of interest.
+
+        The input reference image is expected to be in matrix form and be in
+        either a `H x W` or `H x W x C` shape format.
+
+        Output saliency map matrix should be in the shape `nClasses x H x W`,
+        be floating-point typed and be composed of values in the `[-1, 1]`
+        range.
+        `nClasses` should be the quantity of unique class labels output by the
+        given classifier blackbox.
```

Comment:
Just to check my understanding- this means the blackbox classifier must sub-select class indices (similar to in our current demo notebooks). What are the advantages/disadvantages of this approach vs. passing class indices in the constructor of the `GenerateImageClassifierBlackboxSaliency`? For example, if we wanted to explain new classes, we would need to create a new blackbox classifier instance vs. just changing the class indices param of the constructor.

Reply:
That was my thought here as I didn't think that it was appropriate for this level of functionality to care about that. If that's a common use-case then we can try to support that better here, however I might argue that such index output selection is more appropriate for the blackbox, or a wrapper thereof, to handle.

Reply:
Ok, I think we can revisit this at a later time, but I view the black box as "class-less". The black box could potentially be shared across multiple instantiations of saliency map generation (where the only difference is target class). A wrapper might be a way forward, but you can also see the `target` variable in [Captum's API](https://captum.ai/api/occlusion.html) for reference.

Reply:
I'll take a look at the Captum reference and see what might make sense to do here.